File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.cpp
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer = {arr[0]};
    int last = arr[0];
    
    for (int i=1; i<arr.size(); i++)
    {
        if (arr[i] == last) continue;
        last = arr[i];
        answer.push_back(arr[i]);
    }

    return answer;
}

File: JavaScript/프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.js
function solution(arr)
{
    let last = -1;
    const answer = [];
    
    arr.forEach((e) => 
        {
            if (e != last)
            {
                answer.push(e);
                last = e;
            }
        }
    )
    
    return answer;
}